src/graphic/align.cc

```diff
@@ -76,4 +76,16 @@ Rectf fit_image(float img_w, float img_h, float available_w, float available_h)
 	}
 	return Rectf(0, (available_h - new_h) / 2.f, available_w, new_h);
 }
+
+Rectf fit_image_nocrop(float img_w, float img_h, float available_w, float available_h) {
```

Comment:
The `if` conditions are the only two lines that differ from the `fit_image` function. To reduce code redundancy you can simply use an additional `bool crop` parameter.